src/main/java/roomescape/reservation/ReservationService.java

```diff
@@ -0,0 +1,41 @@
+package roomescape.reservation;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.springframework.stereotype.Service;
+import roomescape.time.ReservationTime;
+import roomescape.time.ReservationTimeRepository;
+
+@Service
+public class ReservationService {
+
+    private final ReservationRepository reservationRepository;
+    private final ReservationTimeRepository reservationTimeRepository;
+
+    public ReservationService(ReservationRepository reservationRepository,
+                              ReservationTimeRepository reservationTimeRepository) {
+        this.reservationRepository = reservationRepository;
+        this.reservationTimeRepository = reservationTimeRepository;
+    }
+
+    public void createReservation(final ReservationRequest request) {
+        final ReservationTime time = reservationTimeRepository.findById(request.timeId());
+        final Reservation reservation = new Reservation(
+                0L,
+                request.name(),
+                request.date(),
+                time
+        );
+        reservationRepository.insert(reservation);
+    }
+
+    public List<Reservation> readAllReservations() {
+        return reservationRepository.findAllReservations();
+    }
+
+    public void deleteReservation(final Long id) {
+        if (reservationRepository.delete(id) == 0) {
+            throw new NoSuchElementException("[ERROR] 해당 ID의 예약을 찾을 수 없습니다. id:" + id);
+        }
```

Comment:
delete의 내부 구현을 알지 못하면 이해하기 힘들 수 있을 것 같아요!
분리해주는 것은 어떤가요?
```suggestion
        int deletedCount = reservationRepository.delete(id)
        if (deletedCount == 0) {
            throw new NoSuchElementException("[ERROR] 해당 ID의 예약을 찾을 수 없습니다. id:" + id);
        }
```